### PR TITLE
Fix MAM+MUC Light

### PR DESCRIPTION
**Pull request by fenek**
This PR addresses #1267 

Proposed changes include:
* A call to `mod_muc_room:can_access_identity` in `mod_mam_muc` has been changed to a hook call. `mod_muc` and `mod_muc_light` have been modified accordingly.
* A short (1ms) sleep has been added in MAM tests, because messages sometimes are stored out of order (because we no longer use strictly monotonic time) on fast machines.

### CODE REVIEW

File: apps/ejabberd/src/mod_mam_muc.erl [hunk 1]
```diff
@@ -325,22 +323,22 @@ is_room_action_allowed_by_default(Action, From, To) ->
...
 
 %% @doc Return true if user element should be removed from results
--spec is_user_identity_hidden(From :: ejabberd:jid(), ArcJID :: ejabberd:jid()) -> boolean().
-is_user_identity_hidden(From, ArcJID) ->
-    case mod_muc_room:can_access_identity(ArcJID, From) of
-        {error, _} -> true;
-        {ok, CanAccess} -> (not CanAccess)
+-spec is_user_identity_hidden(User :: ejabberd:jid(), Room :: ejabberd:jid()) -> boolean().
+is_user_identity_hidden(User, Room) ->
+    case ejabberd_hooks:run_fold(can_access_identity, Room#jid.lserver, false, [Room, User]) of
```
Comment (michalwski) [inline apps/ejabberd/src/mod_mam_muc.erl]: Maybe it'd be good to prefix this hook name with sth indicating this relates to MUC(Light)?

Comment (fenek) [inline apps/ejabberd/src/mod_mam_muc.erl]: `can_access_identities_in_room`? `can_access_identities_in_muc`? `muc_can_access_identities`?

Comment (michalwski) [inline apps/ejabberd/src/mod_mam_muc.erl]: `can_access_identities_in_room` as for me :)

**Comment by michalwski** [comment]
Ok I merge this, as this fixes MAM tests and will stabilise tests on travis. The hook name may still be changed in another MUC related PR.
Thanks!

**Comment by michalwski** [comment]
Also we **need** a regression test for this issue so that we know it doesn't return in the future.

**Pull request merged**